src/App.js

```diff
@@ -30,11 +35,16 @@ class App extends Component {
 
   constructor () {
     super()
-    this.oauthClientId = process.env.REACT_APP_OAUTH_CLIENT_ID
-    this.oauthRedirectURI = process.env.REACT_APP_OAUTH_REDIRECT_URI
 
     this.parsedQueryString = queryString.parse(location.search)
-    let { dataset_name, query, queryType, token } = this.parsedQueryString
+    let { dataset_name, query, queryType, token, code } = this.parsedQueryString
+    const codeVerifier = window.localStorage.getItem('DW-CODE-VERIFIER')
```

Comment:
Is it necessary to do this every time the connector is initialized?  Shouldn't it only be done just before the user enters the auth flow?

Reply:
Yes it should.

Fixed, a new code verifier is now produced right before the user is redirected to the auth URL. This makes checking whether there's a stored code verifier unnecessary.